tests/test_jacobian.py

```diff
@@ -61,36 +64,94 @@ def test_scalar_prm(self):
-        check(jac["out2"]["B"], me.xs)
+        CHECK(jac["out1"]["A"], me.xs)
+        CHECK(jac["out1"]["B"], 2 * B * np.ones_like(me.xs))
+        CHECK(jac["out2"]["A"], me.xs * 2 * A)
+        CHECK(jac["out2"]["B"], me.xs)
 
     def test_vector_prm(self):
         me = DummyMEVectorPrm()
         x = np.asarray(me.parameter_list["X"])
         jac = me.jacobian()
 
         jac_correct = np.concatenate([np.diag(2 * x), np.diag(3 * x ** 2)])
-        np.testing.assert_array_almost_equal(jac["out"]["X"], jac_correct)
+        CHECK(jac["out"]["X"], jac_correct)
 
     def test_vector_prm2(self):
         me = DummyMEVectorPrm2()
         jac = me.jacobian()
 
         jac_correct = np.array([[1, 1]])
-        np.testing.assert_array_almost_equal(jac["out"]["X"], jac_correct)
+        CHECK(jac["out"]["X"], jac_correct)
 
     def test_partial_jacobian_definition(self):
         me = DummyMEPartial()
         A, B = me.parameter_list["A"], me.parameter_list["B"]
         jac = me.jacobian()
-        check = np.testing.assert_array_almost_equal  # just to make it shorter
-        check(jac["out1"]["A"], me.xs)
-        check(jac["out1"]["B"], 2 * B * np.ones_like(me.xs))
-        check(jac["out2"]["A"], me.xs * 2 * A)
-        check(jac["out2"]["B"], me.xs)
+        CHECK(jac["out1"]["A"], me.xs)
+        CHECK(jac["out1"]["B"], 2 * B * np.ones_like(me.xs))
+        CHECK(jac["out2"]["A"], me.xs * 2 * A)
+        CHECK(jac["out2"]["B"], me.xs)
+
+
+class OddEvenME(ModelErrorInterface):
+    def __init__(self):
+        super().__init__()
+        self.parameter_list.define("E_odd", 42.0)
+        self.parameter_list.define("E_even", 4.0)
+        self.parameter_list.define("E_all", 4.0)
+        self.x_odd = np.r_[0, 2, 0, 2, 0, 2]
+        self.x_even = np.r_[1, 0, 1, 0, 1, 0]
+        self.x_all = np.r_[3, 3, 3, 3, 3, 3]
+
+    def __call__(self):
+        return {
+            "out": self.x_odd * self.parameter_list["E_odd"]
+            + self.x_even * self.parameter_list["E_even"]
+            + self.x_all * self.parameter_list["E_all"]
+        }
+
+
+class TestJacobianJointGlobal(unittest.TestCase):
+    def test_individual(self):
+        me = OddEvenME()
+        jac = me.jacobian()
+        CHECK(jac["out"]["E_odd"], me.x_odd)
+        CHECK(jac["out"]["E_even"], me.x_even)
+        CHECK(jac["out"]["E_all"], me.x_all)
+
+    def test_three_joints(self):
+        me = OddEvenME()
+        p = VariationalBayesProblem()
+        p.add_model_error(me)
+        p.latent["E"].add(me.parameter_list, "E_odd")
+        p.latent["E"].add(me.parameter_list, "E_even")
+        p.latent["E"].add(me.parameter_list, "E_all")
+
+        with self.assertRaises(Exception):
+            p.jacobian([42.0])  # we have not defined a noise model yet!
+
+        noise_key = p.add_noise_model(SingleSensorNoise())
+
+        J = p.jacobian([42.0])[noise_key]
+        self.assertEqual(J.shape, (6, 1))
+        CHECK(J[:, 0], -me.x_odd - me.x_even - me.x_all)
```

Comment:
I'm a little confused with the sign here. Maybe this is similar to Abbas issues with what Jacobian we are talking about. But in your model there is no data. Maybe that is part of a discussion, but I personally feel that it should be the Jacobian of the model error, not the forward model (or we provide two different methods).

Reply:
This joining of Jacobians is currently only implemented in `VariationalBayesProblem`. Here, we follow the definition of Chapell where J = -dk/dtheta. Thus the minus. 

But to make that more clear, I propose to move the method *without the minus* to the base class `InferenceProblem` and add the minus only in `VariationalBayesProblem.jacobian`.

Reply:
The problem I see is that even in VB we do not necessarily have to implement a forward model, just the model error. So even if its in line with the paper, the results are applicable when only implementing the model error, so my personal opinion would be to interpret the Jacobian as the derivative of the model error with respect to the parameters, and then use the minus in the update equations directly in VB - maybe by having two variables J (from the paper) = - H (d ME/d theta)

Reply:
That would IMO exceed the scope here. What about just merging this PR now and handle the minus in a separate one?